pkg/apis/config/v1alpha1/tmo.go

```diff
@@ -0,0 +1,144 @@
+type TransparentMemoryOffloadingConfigurationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []TransparentMemoryOffloadingConfiguration `json:"items"`
+}
+
+// TransparentMemoryOffloadingConfigurationSpec defines the desired state of TransparentMemoryOffloadingConfiguration
+type TransparentMemoryOffloadingConfigurationSpec struct {
+	GenericConfigSpec `json:",inline"`
+
+	// Config is custom field for TMO configuration
+	Config TransparentMemoryOffloadingConfig `json:"config"`
+}
+
+type TransparentMemoryOffloadingConfig struct {
+	// QoSLevelConfig is a configuration for manipulating TMO on Different QoS Level
+	// +optional
+	// +listMapKey=qosLevel
+	// +listType=map
+	QoSLevelConfig []QoSLevelConfig `json:"qosConfig,omitempty"`
+
+	// CgroupConfig is a configuration for manipulating TMO on specified cgroups
+	// +optional
+	// +listMapKey=cgroupPath
+	// +listType=map
+	CgroupConfig []CgroupConfig `json:"CgroupConfig"`
+}
+
+type QoSLevelConfig struct {
+	// QoSLevel is either of reclaimed_cores, shared_cores, dedicated_cores, system_cores
+	QoSLevel QoSLevel `json:"qosLevel"`
+
+	// ConfigDetail is configuration details of TMO
+	ConfigDetail TMOConfigDetail `json:"configDetail"`
+}
+
+type QoSLevel string
+
+const (
+	QoSLevelReclaimedCores QoSLevel = "reclaimed_cores"
+	QoSLevelSharedCores    QoSLevel = "shared_cores"
+	QoSLevelDedicatedCores QoSLevel = "dedicated_cores"
+	QoSLevelSystemCores    QoSLevel = "system_cores"
+)
+
+type CgroupConfig struct {
+	// CgroupPath is an cgroupV2 absolute path, e.g. /sys/fs/cgroup/hdfs
+	CgroupPath string `json:"cgroupPath"`
+
+	// ConfigDetail is configuration details of TMO
+	ConfigDetail TMOConfigDetail `json:"configDetail"`
+}
+
+type TMOConfigDetail struct {
+	// EnableTMO is whether to enable TMO on target objective
+	// +optional
+	EnableTMO *bool `json:"enableTMO,omitempty"`
+
+	// EnableSwap is whether to enable swap to offloading anon pages
+	// +optional
+	EnableSwap *bool `json:"enableSwap,omitempty"`
+
+	// Interval is the minimum duration the objectives got memory reclaimed by TMO
+	// +optional
+	Interval *metav1.Duration `json:"interval,omitempty"`
+
+	// PolicyName is used to specify the policy for calculating memory offloading size
+	PolicyName TMOPolicyName `json:"policyName"`
+
+	// PsiPolicyConf is configurations of a TMO policy which reclaim memory by PSI
+	// +optional
+	PsiPolicyConf *PsiPolicyConf `json:"psiPolicy,omitempty"`
+
+	// RefaultPolicy is configurations of a TMO policy which reclaim memory by refault
+	// +optional
+	RefaultPolicConf *RefaultPolicyConf `json:"refaultPolicy,omitempty"`
+}
+
+type TMOPolicyName string
+
+const (
+	DefaultPolicy TMOPolicyName = "default_policy"
+	PsiPolicy     TMOPolicyName = "psi_policy"
+	RefaultPolicy TMOPolicyName = "refault_policy"
+)
+
+type PsiPolicyConf struct {
+	MaxProbe          float64 `json:"maxProbe"`
```

Comment:
whether all the conf should have default value? It may use pointer and omitempty more properly